### ISSUE TITLE
Crash in TDB when making an HTTP request

### ISSUE THREAD

**Issue by keshonok** [natsys/tempesta#108]
This is done on top of ab-startfix and ab-locks branches, as current master branch is not quite functional yet.

Put a server outside of any group in Tempesta configuration, so that requests are forwarded to that server. Cache is enabled by default now. Start Tempesta, and make an HTTP request with `wget`. Get the crash.

Internally, Tempesta makes an attempt to send a response to HTTP request from it's cache (TDB). That's where the crash happens.

```
Loading Tempesta kernel modules...
[   59.570088] [tdb] Start Tempesta DB
[   59.751535] [tempesta] Initializing Tempesta FW kernel module...
[   59.752516] [tempesta] init: cfg_if
[   59.753076] [tempesta] init: tls
[   59.753581] [tempesta] init: http
[   59.754093] [tempesta] init: http_sticky
[   59.757210] [tempesta] init: server
[   59.757788] [tempesta] init: client
[   59.758344] [tempesta] init: sock_srv
[   59.758866] [tempesta] init: sock_clnt
[   59.759226] [tempesta]   register cfg: filter
[   59.759627] [tempesta]   register cfg: cache
[   59.760039] [tempesta]   register cfg: http_sticky
[   59.760475] [tempesta]   register cfg: sock_srv
[   59.760890] [tempesta]   register cfg: sock_clnt
[   59.776664] [tempesta] Registering new scheduler: dummy
[   59.793796] [tempesta]   tfw_sched_hash: init
[   59.794222] [tempesta] Registering new scheduler: hash
[   59.809295] [tempesta]   tfw_sched_http: init
[   59.809722] [tempesta]   register cfg: tfw_sched_http
[   59.810192] [tempesta] Registering new scheduler: http
[   59.825986] [tempesta]   tfw_sched_rr: init
[   59.826392] [tempesta] Registering new scheduler: round-robin
Starting Tempesta...
[   59.830024] [tempesta] got state via sysctl: start
[   59.830406] [tempesta]   reading configuration file...
[   59.830792] [tempesta]   reading file: /home/natsys-lab/tfw-run/tempesta/etc/tempesta_fw.conf
[   59.833118] [tempesta]   file size: 5772 bytes
[   59.833465] [tempesta]   read by offset: 0
[   59.833810] [tempesta]   read by offset: 4096
[   59.834174] [tempesta]   read by offset: 5772
[   59.834527] [tempesta] starting all modules...
[   59.834856] [tempesta]   parsing configuration and pushing it to modules...
[   59.835404] [tempesta]   spec handle: 'server'
[   59.835944] [tempesta]   new server group: 'default'
[   59.836960] [tempesta]   use default entry: 'filter_tbl_size 16777216;'
[   59.837639] [tempesta]   spec handle: 'filter_tbl_size'
[   59.838127] [tempesta]   use default entry: 'filter_db /opt/tempesta/db/filter.tdb;'
[   59.838849] [tempesta]   spec handle: 'filter_db'
[   59.839291] [tempesta]   use default entry: 'cache on;'
[   59.839782] [tempesta]   spec handle: 'cache'
[   59.840218] [tempesta]   use default entry: 'cache_size 268435456;'
[   59.840896] [tempesta]   spec handle: 'cache_size'
[   59.841334] [tempesta]   use default entry: 'cache_db /opt/tempesta/db/cache.tdb;'
[   59.841856] [tempesta]   spec handle: 'cache_db'
[   59.842289] [tempesta]   use default entry: 'listen 80;'
[   59.842793] [tempesta]   spec handle: 'listen'
[   59.843433] [tempesta]   starting modules...
[   59.843835] [tempesta]   mod_start(): filter
[   59.850879] [tdb] Opened table /opt/tempesta/db/filter.tdb: size=16777216 rec_size=20 base=ffff880033000000
[   59.851833] [tempesta]   mod_start(): cache
[   59.972801] [tdb] Opened table /opt/tempesta/db/cache.tdb: size=268435456 rec_size=0 base=ffff880023000000
[   59.973883] [tempesta]   mod_start(): http_sticky
[   59.974328] [tempesta]   mod_start(): sock_srv
[   59.974812] BUG: sleeping function called from invalid context at net/core/sock.c:2286
[   59.975549] in_atomic(): 1, irqs_disabled(): 0, pid: 2376, name: sysctl
[   59.976175] 4 locks held by sysctl/2376:
[   59.976543]  #0:  (sb_writers#6){.+.+.+}, at: [<ffffffff810e9194>] vfs_write+0xb4/0x1b0
[   59.977355]  #1:  (tfw_cfg_sysctl_state_buf_mtx){+.+.+.}, at: [<ffffffffa00110ba>] handle_sysctl_state_io+0x54/0x1bb [tempesta_fw]
[   59.978486]  #2:  (sg_lock){++++..}, at: [<ffffffffa004cf6a>] tfw_sg_for_each_srv+0x1f/0xfa [tempesta_fw]
[   59.979431]  #3:  (&sg->lock){+.+...}, at: [<ffffffffa004cf92>] tfw_sg_for_each_srv+0x47/0xfa [tempesta_fw]
[   59.980395] CPU: 0 PID: 2376 Comm: sysctl Tainted: G           O 3.10.10-tempesta #2
[   59.981109]  0000000000000948 ffff880038851ba8 ffffffff813c5eed ffff880038851bc8
[   59.981835]  ffffffff8105ba47 ffff880038af6940 ffff880038af6990 ffff880038851bf8
[   59.982563]  ffffffff812dbf6a ffff880000000000 ffff880038851df0 000000000000001f
[   59.983291] Call Trace:
[   59.983529]  [<ffffffff813c5eed>] dump_stack+0x19/0x1b
[   59.984018]  [<ffffffff8105ba47>] __might_sleep+0x117/0x130
[   59.984531]  [<ffffffff812dbf6a>] lock_sock_nested+0x2a/0x90
[   59.985053]  [<ffffffffa004d9eb>] lock_sock+0x1d/0x1f [tempesta_fw]
[   59.985636]  [<ffffffffa004fdc0>] ss_connect+0x80/0xe3 [tempesta_fw]
[   59.986222]  [<ffffffffa0050d8f>] tfw_sock_srv_connect_try+0xb3/0xf6 [tempesta_fw]
[   59.986919]  [<ffffffffa00511b2>] tfw_sock_srv_connect_srv+0x31/0x6f [tempesta_fw]
[   59.987622]  [<ffffffffa004cfb9>] tfw_sg_for_each_srv+0x6e/0xfa [tempesta_fw]
[   59.988293]  [<ffffffffa0051181>] ? tfw_sock_srv_disconnect+0x9d/0x9d [tempesta_fw]
[   59.989002]  [<ffffffffa005126b>] tfw_sock_srv_connect_all+0x10/0x12 [tempesta_fw]
[   59.989698]  [<ffffffffa0010656>] mod_start+0x6e/0x9b [tempesta_fw]
[   59.990282]  [<ffffffffa0010ac8>] tfw_cfg_start_mods+0xf0/0x1f6 [tempesta_fw]
[   59.990942]  [<ffffffffa0010fd1>] handle_state_change+0xfd/0x192 [tempesta_fw]
[   59.991604]  [<ffffffffa00111b6>] handle_sysctl_state_io+0x150/0x1bb [tempesta_fw]
[   59.992144]  [<ffffffffa00110ba>] ? handle_sysctl_state_io+0x54/0x1bb [tempesta_fw]
[   59.992833]  [<ffffffff811408bf>] proc_sys_call_handler+0x8f/0xc0
[   59.993390]  [<ffffffff811408bf>] ? proc_sys_call_handler+0x8f/0xc0
[   59.993967]  [<ffffffff811408ff>] proc_sys_write+0xf/0x20
[   59.994467]  [<ffffffff810e91b3>] vfs_write+0xd3/0x1b0
[   59.994944]  [<ffffffff810e9787>] SyS_write+0x47/0x90
[   59.995408]  [<ffffffff813cb2a9>] system_call_fastpath+0x16/0x1b
[   59.995958] BUG: scheduling while atomic: sysctl/2376/0x10000003
[   59.996523] 4 locks held by sysctl/2376:
[   59.996887]  #0:  (sb_writers#6){.+.+.+}, at: [<ffffffff810e9194>] vfs_write+0xb4/0x1b0
[   59.997684]  #1:  (tfw_cfg_sysctl_state_buf_mtx){+.+.+.}, at: [<ffffffffa00110ba>] handle_sysctl_state_io+0x54/0x1bb [tempesta_fw]
[   59.998817]  #2:  (sg_lock){++++..}, at: [<ffffffffa004cf6a>] tfw_sg_for_each_srv+0x1f/0xfa [tempesta_fw]
[   59.999755]  #3:  (&sg->lock){+.+...}, at: [<ffffffffa004cf92>] tfw_sg_for_each_srv+0x47/0xfa [tempesta_fw]
[   60.000707] Modules linked in: tfw_sched_rr(O) tfw_sched_http(O) tfw_sched_hash(O) tfw_sched_dummy(O) tempesta_fw(O) tempesta_db(O)
[   60.001858] CPU: 0 PID: 2376 Comm: sysctl Tainted: G           O 3.10.10-tempesta #2
[   60.002566]  ffff880038af6990 ffff880038851b28 ffffffff813c5eed ffff880038851b40
[   60.003288]  ffffffff813c2cc3 ffff88003fc11700 ffff880038851ba0 ffffffff813c7f90
[   60.004001]  0000000000011700 ffff880038851fd8 ffff880038851fd8 0000000000011700
[   60.004726] Call Trace:
[   60.004958]  [<ffffffff813c5eed>] dump_stack+0x19/0x1b
[   60.005429]  [<ffffffff813c2cc3>] __schedule_bug+0x5f/0x6e
[   60.005930]  [<ffffffff813c7f90>] __schedule+0x90/0x620
[   60.006409]  [<ffffffff8105c2f1>] __cond_resched+0x21/0x30
[   60.006922]  [<ffffffff813c88e2>] _cond_resched+0x32/0x50
[   60.007420]  [<ffffffff812dbf6f>] lock_sock_nested+0x2f/0x90
[   60.007944]  [<ffffffffa004d9eb>] lock_sock+0x1d/0x1f [tempesta_fw]
[   60.008537]  [<ffffffffa004fdc0>] ss_connect+0x80/0xe3 [tempesta_fw]
[   60.009138]  [<ffffffffa0050d8f>] tfw_sock_srv_connect_try+0xb3/0xf6 [tempesta_fw]
[   60.009834]  [<ffffffffa00511b2>] tfw_sock_srv_connect_srv+0x31/0x6f [tempesta_fw]
[   60.010522]  [<ffffffffa004cfb9>] tfw_sg_for_each_srv+0x6e/0xfa [tempesta_fw]
[   60.011167]  [<ffffffffa0051181>] ? tfw_sock_srv_disconnect+0x9d/0x9d [tempesta_fw]
[   60.011869]  [<ffffffffa005126b>] tfw_sock_srv_connect_all+0x10/0x12 [tempesta_fw]
[   60.012532]  [<ffffffffa0010656>] mod_start+0x6e/0x9b [tempesta_fw]
[   60.013074]  [<ffffffffa0010ac8>] tfw_cfg_start_mods+0xf0/0x1f6 [tempesta_fw]
[   60.013682]  [<ffffffffa0010fd1>] handle_state_change+0xfd/0x192 [tempesta_fw]
[   60.014301]  [<ffffffffa00111b6>] handle_sysctl_state_io+0x150/0x1bb [tempesta_fw]
[   60.014959]  [<ffffffffa00110ba>] ? handle_sysctl_state_io+0x54/0x1bb [tempesta_fw]
[   60.015617]  [<ffffffff811408bf>] proc_sys_call_handler+0x8f/0xc0
[   60.016150]  [<ffffffff811408bf>] ? proc_sys_call_handler+0x8f/0xc0
[   60.016691]  [<ffffffff811408ff>] proc_sys_write+0xf/0x20
[   60.017159]  [<ffffffff810e91b3>] vfs_write+0xd3/0x1b0
[   60.017605]  [<ffffffff810e9787>] SyS_write+0x47/0x90
[   60.018044]  [<ffffffff813cb2a9>] system_call_fastpath+0x16/0x1b
[   60.018823] [tempesta]   mod_start(): sock_clnt
[   60.019404] [sync_sockets]   ss_tcp_state_change: sk ffff880038af6940, sk->sk_socket           (null), state (Established)
[   60.020423] [tempesta]   connected: 192.168.10.1:8080
[   60.020911] [sync_sockets]   ss_tcp_state_change: sk ffff880038d1b280, sk->sk_socket           (null), state (Established)
[   60.021962] [tempesta]   connected: 192.168.10.1:8080
[   60.022465] [sync_sockets]   ss_tcp_state_change: sk ffff880038d1a980, sk->sk_socket           (null), state (Established)
[   60.023618] [tempesta]   connected: 192.168.10.1:8080
[   60.024178] [sync_sockets]   ss_tcp_state_change: sk ffff880038d1a080, sk->sk_socket           (null), state (Established)
[   60.025314] [tempesta]   connected: 192.168.10.1:8080
[   60.025887] [tempesta] Open listen socket on: 0.0.0.0
[   60.026577] [tempesta]   start listening on socket: sk=ffff88003862b2c0
[   60.027323] [tempesta]   mod_start(): tfw_sched_http
[   60.027858] [tempesta] modules are started
net.tempesta.state = start
done
done
[  108.254299] [sync_sockets]   ss_tcp_state_change: sk ffff88003862a0c0, sk->sk_socket           (null), state (Established)
[  108.255828] [tempesta]   new client socket: sk=ffff88003862a0c0, state=1
[  108.256603] [tempesta]   new client: cli=ffff880038a9ba70
[  108.257295] [tempesta]   new client socket is accepted: sk=ffff88003862a0c0, conn=ffff8800384ef6e8, cli=ffff880038a9ba70
[  108.258037] [sync_sockets]   ss_drain_accept_queue: sk ffff88003862b2c0, sk->sk_socket           (null), state (Listen)
[  108.258037] [sync_sockets]   ss_tcp_data_ready: sk ffff88003862a0c0, sk->sk_socket           (null), state (Established)
[  108.258037] [tempesta]   Link new msg ffff880038540040 with connection ffff8800384ef6e8
[  108.258037] [tempesta]   Add skb ffff880038ac03c0 to message ffff880038540040
[  108.258037] [tempesta]   Received 62 client data bytes (GET / HTTP/1.1
[  108.258037] User-Agent: Wget/1.15 (linux-gnu)
[  108.258037] Accept: */*) on socket (conn=ffff8800384ef6e8)
[  108.258037] [tempesta]   enter FSM at state 0
[  108.258037] [tempesta]   parser: Req_0(0:0): c=0x47(G), r=-2
[  108.258037] [tempesta]   parser: Req_Method(1:0): c=0x47(G), r=-2
[  108.258037] [tempesta]   parser: Req_MUSpace(2:0): c=0x2f(/), r=-2
[  108.258037] [tempesta]   parser: Req_UriAbsPath(6:0): c=0x20( ), r=-2
[  108.258037] [tempesta]   parser: Req_HttpVer(7:0): c=0x48(H), r=-2
[  108.258037] [tempesta]   parser: Req_EoL(15:0): c=0xd(.), r=-2
[  108.258037] [tempesta]   parser: Req_EoL(15:0): c=0xa(.), r=-2
[  108.258037] [tempesta]   parser: Req_Hdr(16:0): c=0x55(U), r=-2
[  108.258037] [tempesta]   parser: Req_HdrOther(92:0): c=0x55(U), r=-2
[  108.258037] [tempesta]   store header w/ ptr=ffff880038cfa092 len=33 flags=0 id=3
[  108.258037] [tempesta]   parser: Req_Hdr(16:0): c=0x41(A), r=-2
[  108.258037] [tempesta]   parser: Req_HdrOther(92:0): c=0x41(A), r=-2
[  108.258037] [tempesta]   store header w/ ptr=ffff880038cfa0b5 len=11 flags=20 id=4
[  108.258037] [tempesta]   request parsed: len=62 parsed=62 msg_len=62 res=-1
[  108.258037] [tempesta]   GFSM return code 0
[  108.258037] [tempesta]   postpone skb ffff880038ac03c0
[  108.258037] [tempesta]   Received 50 client data bytes (
[  108.258037] Host: 192.168.10.230
[  108.258037] Connection: Keep-Alive
[  108.258037] 
[  108.258037] ) on socket (conn=ffff8800384ef6e8)
[  108.258037] [tempesta]   enter FSM at state 92
[  108.258037] [tempesta]   parser: Req_HdrOther(92:0): c=0xd(.), r=-2
[  108.258037] [tempesta]   store header w/ ptr=          (null) len=0 flags=0 id=4
[  108.258037] [tempesta]   parser: Req_Hdr(16:0): c=0x48(H), r=-2
[  108.258037] [tempesta]   parser: RGen_LWS(10000:21): c=0x20( ), r=-2
[  108.258037] [tempesta]   parser: RGen_LWS(10000:21): c=0x31(1), r=-2
[  108.258037] [tempesta]   enter FSM at state 21
[  108.258037] [tempesta]   parser: Req_HdrHostV(21:0): c=0x31(1), r=-2
[  108.258037] [tempesta]   enter FSM at state 1
[  108.258037] [tempesta]   parser: Req_I_H(1:1): c=0x31(1), r=-2
[  108.258037] [tempesta]   parser: Req_I_H(1:1): c=0x39(9), r=-2
[  108.258037] [tempesta]   parser: Req_I_H(1:1): c=0x32(2), r=-2
[  108.258037] [tempesta]   parser: Req_I_H(1:1): c=0x2e(.), r=-2
[  108.258037] [tempesta]   parser: Req_I_H(1:1): c=0x31(1), r=-2
[  108.258037] [tempesta]   parser: Req_I_H(1:1): c=0x36(6), r=-2
[  108.258037] [tempesta]   parser: Req_I_H(1:1): c=0x38(8), r=-2
[  108.258037] [tempesta]   parser: Req_I_H(1:1): c=0x2e(.), r=-2
[  108.258037] [tempesta]   parser: Req_I_H(1:1): c=0x31(1), r=-2
[  108.258037] [tempesta]   parser: Req_I_H(1:1): c=0x30(0), r=-2
[  108.258037] [tempesta]   parser: Req_I_H(1:1): c=0x2e(.), r=-2
[  108.258037] [tempesta]   parser: Req_I_H(1:1): c=0x32(2), r=-2
[  108.258037] [tempesta]   parser: Req_I_H(1:1): c=0x33(3), r=-2
[  108.258037] [tempesta]   parser: Req_I_H(1:1): c=0x30(0), r=-2
[  108.258037] [tempesta]   parser: Req_I_H(1:1): c=0xd(.), r=-2
[  108.258037] [tempesta]   parser: Req_I_H_EoL(3:1): c=0xd(.), r=-2
[  108.258037] [tempesta]   parser: Req_I_H_EoL(3:1): c=0xa(.), r=-2
[  108.258037] [tempesta]   parse header __req_parse_host: ret=16 plen=14 id=1
[  108.258037] [tempesta]   store header w/ ptr=ffff88003833f08e len=20 flags=0 id=1
[  108.258037] [tempesta]   parser: Req_Hdr(16:0): c=0x43(C), r=-2
[  108.258037] [tempesta]   parser: Req_HdrC(22:0): c=0x6f(o), r=-2
[  108.258037] [tempesta]   parser: Req_HdrConnection(44:0): c=0x3a(:), r=-2
[  108.258037] [tempesta]   parser: RGen_LWS(10000:45): c=0x20( ), r=-2
[  108.258037] [tempesta]   parser: RGen_LWS(10000:45): c=0x4b(K), r=-2
[  108.258037] [tempesta]   enter FSM at state 45
[  108.258037] [tempesta]   parser: Req_HdrConnectionV(45:0): c=0x4b(K), r=-2
[  108.258037] [tempesta]   enter FSM at state 1
[  108.258037] [tempesta]   parser: I_Conn(1:1): c=0x4b(K), r=-2
[  108.258037] [tempesta]   parser: I_EoT(6:1): c=0xd(.), r=0
[  108.258037] [tempesta]   parser: I_EoL(7:1): c=0xd(.), r=0
[  108.258037] [tempesta]   parser: I_EoL(7:1): c=0xa(.), r=0
[  108.258037] [tempesta]   parser: Connection parsed: flags 0x2
[  108.258037] [tempesta]   parse header __parse_connection: ret=12 plen=10 id=0
[  108.258037] [tempesta]   store header w/ ptr=ffff88003833f0a4 len=22 flags=0 id=0
[  108.258037] [tempesta]   parser: Req_Hdr(16:0): c=0xd(.), r=-2
[  108.258037] [tempesta]   parser: Req_HdrDone(93:0): c=0xa(.), r=-2
[  108.258037] [tempesta]   parse msg body: flags=0x2 content_length=0
[  108.258037] [tempesta]   request parsed: len=50 parsed=50 msg_len=50 res=0
[  108.258037] [tempesta]   GFSM return code 0
[  108.258037] [tempesta]   Matching request: ffff880038540040, list: ffff880038985040
[  108.258037] [tempesta]   tfw_sched_http: use server group: 'default'
[  108.258037] [tempesta] ERROR: can't add X-Forwarded-For header to msg: ffff880038540040, buf: X-Forwarded-For: 192.168.10.1
[  108.258037] [tempesta]   Free msg: ffff880038540040
[  108.258037] [tempesta]   free skb ffff880038ac03c0: truesize=4864 sk=ffff88003862a0c0, destructor=ffffffff812dab30 users=1 type=Conn_Clnt
[  108.258037] ------------[ cut here ]------------
[  108.258037] kernel BUG at /home/natsys-lab/tfw-run/tempesta/tempesta_db/core/main.c:108!
[  108.258037] invalid opcode: 0000 [#1] SMP 
[  108.258037] Modules linked in: tfw_sched_rr(O) tfw_sched_http(O) tfw_sched_hash(O) tfw_sched_dummy(O) tempesta_fw(O) tempesta_db(O)
[  108.258037] CPU: 0 PID: 0 Comm: swapper/0 Tainted: G        W  O 3.10.10-tempesta #2
[  108.258037] task: ffffffff81565440 ti: ffffffff81554000 task.ti: ffffffff81554000
[  108.258037] RIP: 0010:[<ffffffffa0002550>]  [<ffffffffa0002550>] tdb_rec_put+0x20/0x30 [tempesta_db]
[  108.258037] RSP: 0018:ffff88003fc03910  EFLAGS: 00010246
[  108.258037] RAX: 0000000000000000 RBX: ffff88003862a0c0 RCX: 0000000000000000
[  108.258037] RDX: 0000000000000003 RSI: ffffffff81565a20 RDI: 0000000000000000
[  108.258037] RBP: ffff88003fc03948 R08: 0000000000000000 R09: 0000000000000000
[  108.258037] R10: 0000000000000001 R11: 0000000000000004 R12: ffff880038ac03c0
[  108.258037] R13: ffff88003862a8a0 R14: 0000000000000000 R15: ffff88003862a8a0
[  108.258037] FS:  0000000000000000(0000) GS:ffff88003fc00000(0000) knlGS:0000000000000000
[  108.258037] CS:  0010 DS: 0000 ES: 0000 CR0: 000000008005003b
[  108.258037] CR2: 0000000000609a80 CR3: 0000000038ac7000 CR4: 00000000000006b0
[  108.258037] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[  108.258037] DR3: 0000000000000000 DR6: 00000000ffff0ff0 DR7: 0000000000000400
[  108.258037] Stack:
[  108.258037]  ffffffffa000dd19 ffff8800388f3a48 ffffffffa00169b6 00000000625ae083
[  108.258037]  ffff880038540040 0000000000000000 0000000000000000 ffff88003fc03990
[  108.258037]  ffffffffa000decd ffff8800388f3a48 ffffffffa00169b6 ffff880038540040
[  108.258037] Call Trace:
[  108.258037]  <IRQ> 
[  108.258037]  [<ffffffffa000dd19>] ? __cache_req_process_node+0x9b/0x9d [tempesta_fw]
[  108.258037]  [<ffffffffa00169b6>] ? tfw_http_adjust_resp+0x13/0x13 [tempesta_fw]
[  108.258037]  [<ffffffffa000decd>] tfw_cache_req_process+0x15c/0x163 [tempesta_fw]
[  108.258037]  [<ffffffffa00169b6>] ? tfw_http_adjust_resp+0x13/0x13 [tempesta_fw]
[  108.258037]  [<ffffffffa0016d56>] tfw_http_req_process+0x341/0x3b0 [tempesta_fw]
[  108.258037]  [<ffffffff811fc527>] ? __dynamic_pr_debug+0x97/0xc0
[  108.258037]  [<ffffffffa0017203>] tfw_http_msg_process+0x43/0x5e [tempesta_fw]
[  108.258037]  [<ffffffffa0013118>] tfw_gfsm_dispatch+0x41/0x43 [tempesta_fw]
[  108.258037]  [<ffffffffa0011ea1>] tfw_connection_recv+0x3a/0x3c [tempesta_fw]
[  108.258037]  [<ffffffffa004e37d>] ss_tcp_process_proto_skb+0x74/0xf2 [tempesta_fw]
[  108.258037]  [<ffffffffa004e5a3>] ss_tcp_process_skb+0x1a8/0x270 [tempesta_fw]
[  108.258037]  [<ffffffffa004f06d>] ss_tcp_process_data+0x12c/0x2bb [tempesta_fw]
[  108.258037]  [<ffffffffa004f40c>] ss_tcp_data_ready+0xd1/0x149 [tempesta_fw]
[  108.258037]  [<ffffffff8133beed>] tcp_data_queue+0x4dd/0xd10
[  108.258037]  [<ffffffff8133f2f0>] tcp_rcv_established+0x560/0x620
[  108.258037]  [<ffffffff813494cc>] tcp_v4_do_rcv+0x1dc/0x4e0
[  108.258037]  [<ffffffff811f66b2>] ? do_raw_spin_lock+0x82/0x140
[  108.258037]  [<ffffffff8134b077>] tcp_v4_rcv+0x437/0x8b0
[  108.258037]  [<ffffffff81325455>] ip_local_deliver_finish+0xb5/0x150
[  108.258037]  [<ffffffff813253e0>] ? ip_local_deliver_finish+0x40/0x150
[  108.258037]  [<ffffffff81325c00>] ip_local_deliver+0x50/0x90
[  108.258037]  [<ffffffff8132596e>] ip_rcv_finish+0x47e/0x580
[  108.258037]  [<ffffffff81325efc>] ip_rcv+0x2bc/0x340
[  108.258037]  [<ffffffff812ee4f9>] __netif_receive_skb_core+0x469/0x590
[  108.258037]  [<ffffffff812ee0f3>] ? __netif_receive_skb_core+0x63/0x590
[  108.258037]  [<ffffffff812ee66f>] __netif_receive_skb+0x4f/0x70
[  108.258037]  [<ffffffff812ef1d3>] netif_receive_skb+0xb3/0x140
[  108.258037]  [<ffffffff812e1f09>] ? __netdev_alloc_skb+0x49/0xb0
[  108.258037]  [<ffffffff81278c44>] virtnet_poll+0x624/0x770
[  108.258037]  [<ffffffff812f0413>] net_rx_action+0xa3/0x1f0
[  108.258037]  [<ffffffff81038220>] __do_softirq+0xe0/0x1d0
[  108.258037]  [<ffffffff81038499>] irq_exit+0x69/0xc0
[  108.258037]  [<ffffffff81003c5d>] do_IRQ+0x9d/0xc0
[  108.258037]  [<ffffffff813ca972>] common_interrupt+0x72/0x72
[  108.258037]  <EOI> 
[  108.258037]  [<ffffffff81023626>] ? native_safe_halt+0x6/0x10
[  108.258037]  [<ffffffff8100a8dd>] default_idle+0x1d/0x30
[  108.258037]  [<ffffffff8100b26a>] arch_cpu_idle+0xa/0x10
[  108.258037]  [<ffffffff8106817c>] cpu_startup_entry+0x15c/0x200
[  108.258037]  [<ffffffff813be53d>] rest_init+0xcd/0xe0
[  108.258037]  [<ffffffff813be470>] ? csum_partial_copy_generic+0x170/0x170
[  108.258037]  [<ffffffff815dada5>] start_kernel+0x352/0x35d
[  108.258037]  [<ffffffff815da85c>] ? repair_env_string+0x5e/0x5e
[  108.258037]  [<ffffffff815da120>] ? early_idt_handlers+0x120/0x120
[  108.258037]  [<ffffffff815da55b>] x86_64_start_reservations+0x2a/0x2c
[  108.258037]  [<ffffffff815da64c>] x86_64_start_kernel+0xef/0xfc
[  108.258037] Code: 0f 0b 66 0f 1f 84 00 00 00 00 00 48 83 c7 7f 48 83 e7 80 74 16 55 48 83 c7 08 48 89 e5 e8 b9 79 3c e1 5d c3 0f 1f 80 00 00 00 00 <0f> 0b 66 66 66 66 66 2e 0f 1f 84 00 00 00 00 00 55 48 89 e5 53 
[  108.258037] RIP  [<ffffffffa0002550>] tdb_rec_put+0x20/0x30 [tempesta_db]
[  108.258037]  RSP <ffff88003fc03910>
[  108.258037] ---[ end trace 0431b42c467880a8 ]---
[  108.258037] Kernel panic - not syncing: Fatal exception in interrupt
```

**Comment by keshonok** [comment]
Fixed in 27196d7

**Comment by keshonok** [comment]
Merged to master.
